// @delon/acl/index.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
// region: import
import { ACLService } from './services/acl.service';
var SERVICES = [ACLService];
// components
import { ACLDirective } from './directives/acl.directive';
var COMPONENTS = [ACLDirective];
// pipes
var PIPES = [];
// endregion
// region: export
export { ACLService } from './services/acl.service';
export { ACLDirective } from './directives/acl.directive';
// components
// endregion
var AlainACLModule = /** @class */ (function () {
    function AlainACLModule() {
    }
    AlainACLModule.forRoot = function () {
        return {
            ngModule: AlainACLModule,
            providers: SERVICES.slice()
        };
    };
    AlainACLModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule
                    ],
                    declarations: COMPONENTS.concat(PIPES),
                    exports: COMPONENTS.concat(PIPES)
                },] },
    ];
    /** @nocollapse */
    AlainACLModule.ctorParameters = function () { return []; };
    return AlainACLModule;
}());
export { AlainACLModule };
